### ISSUE TITLE
andThen() test helper doesn't wait for bs-modal's closedAction

### ISSUE THREAD

**Issue by omarkohl** [kaliber5/ember-bootstrap#53]
I'm writing an acceptance test for an application using *bs-modal*. The problem is that the *andThen()* test helper does not wait for the *closedAction()* to be executed when *fade=true* (as is the default).

Probably this is because Bootstrap's *emulateTransitionEnd* calls *setTimeout()* and the *andThen()* helper is unable to cope with this (not being a promise).

What are my options? Adding a delay (either with *setTimeout* or with *Ember.run.later*) in the test seems to do the trick. But I'm wondering if there is a better way. Are there any examples or documentation concerning tests and ember-bootstrap?

Below a full example. A modal dialog is opened. Once the dialog is closed the background-color of the body is set to red. The acceptance test will fail (because it tests for background-color before the color is changed). When settings fade=false the test succeeds:

**app/templates/application.hbs**

```handlebars
{{my-popup}}  
```

**app/components/my-popup.js**
```javascript
import Ember from 'ember';

export default Ember.Component.extend({
  actions: {
    closed() {
      console.log('Modal completely closed.');
      Ember.$('body').css('background-color', 'red');
    }
  }
});
```

**app/components/my-popup.hbs**
```handlebars
{{#bs-modal closedAction='closed' fade=true}}
  I'm a popup!
{{/bs-modal}}
```

**tests/acceptance/test-my-popup-test.js**
```javascript
import { test } from 'qunit';
import Ember from 'ember';
import moduleForAcceptance from 'ember-bs/tests/helpers/module-for-acceptance';

moduleForAcceptance('Acceptance | test popup');

test('popup changes background', function(assert) {
  visit('/');
  click('.modal-dialog button[type="button"]');
  andThen(function() {
    var bodyColor = Ember.$('body').css('background-color');
    assert.strictEqual(bodyColor, 'rgb(255, 0, 0)');
  });
});
```

**Comment by simonihmig** [comment]
You are right, this needs to be fixed. 

I actually had an initial implementation using the original bootstrap.js modal implementation wrapped in a simple component, and that suffered from the same problem you mentioned: the setTimeout call does not play nicely with ember's acceptance tests. I thought I had eliminated all those setTimeout calls by having a native Ember implementation, but you are right: in emulateTransitionEnd imported form the original bootstrap library there is one left. Hope to get that fixed soon...

**Comment by simonihmig** [comment]
Fixed in v0.6.2
